File: tests/interfile/cpp/test_hof_comprehensive_cpp_split/test_original_example.cpp
void test_original_example() {
    std::string history = getHistory("name", "owner");
    std::vector<std::string> vec = {history};
    customForEach<std::string>(vec, [](std::string node) {
        std::string changes = node;
        // ruleid: test-hof-taint
        sink(changes);
    });
}

File: tests/interfile/rust/test_hof_comprehensive_rust_split/test_original_example.rs
use crate::get_history::get_history;
use crate::sink::sink;
use crate::zz_main::custom_for_each;

fn test_original_example() {
    let history = get_history("name", "owner");
    let arr = vec![history];
    custom_for_each(&arr, |node| {
        let changes = node.clone();
        // ruleid: test-hof-taint
        sink(changes);
    });
}
